# [\#14359 Issue](https://github.com/ethereum/solidity/issues/14359) `open`: Unreachable Code warning for abstract contract
**Labels**: `bug :bug:`, `low effort`, `low impact`


#### <img src="https://avatars.githubusercontent.com/u/103474701?u=e22a6306e4d84b001819a6048e91b3624c23577b&v=4" width="50">[frontier159](https://github.com/frontier159) opened issue at [2023-06-26 08:28](https://github.com/ethereum/solidity/issues/14359):

## Description

When an abstract contract function calls a pure virtual function, and a concrete implementation of that function (in a concrete contracts) reverts, the `Warning: Unreachable code.` is shown which is incorrect/misleading.

## Environment

- Compiler version: commit https://github.com/ethereum/solidity/commit/a1b79de64235f13e6b06e088fe6365c5a12d13d3

## Steps to Reproduce

```solidity
pragma solidity ^0.8.20;
// SPDX-License-Identifier: AGPL-3.0-or-later

abstract contract AbstractContract {
    bool public abstractCalled;
    error Unimplemented();

    function outer() external {
        inner();

        abstractCalled = true;
    }

    function inner() internal virtual;
}

contract ChildContract1 is AbstractContract {
    bool public wasCalled;

    function inner() internal override {
        wasCalled = true;
    }
}

contract ChildContract2 is AbstractContract {

    function inner() internal pure override {
        revert Unimplemented();
    }
}
```

Gives the error:
```
Warning: Unreachable code.
  --> contracts/6_unreachable.sol:11:9:
   |
11 |         abstractCalled = true;
   |         ^^^^^^^^^^^^^^^^^^^^^
```

However when `ChildContract1.outer()` is called, the `wasCalled` is correctly set to true. The warning is misleading.





-------------------------------------------------------------------------------



[Export of Github issue for [ethereum/solidity](https://github.com/ethereum/solidity). Generated on 2024.12.15 at 06:45:24.]
